lib/ansible/modules/windows/win_domain.ps1

```diff
@@ -29,17 +29,28 @@ $domain_netbios_name = Get-AnsibleParam -obj $params -name "domain_netbios_name"
 $safe_mode_admin_password = Get-AnsibleParam -obj $params -name "safe_mode_password" -failifempty $true
 $database_path = Get-AnsibleParam -obj $params -name "database_path" -type "path"
 $sysvol_path = Get-AnsibleParam -obj $params -name "sysvol_path" -type "path"
+$create_dns_delegation = Get-AnsibleParam -obj $params -name "create_dns_delegation" -type "bool" -default $true
+$domain_mode = Get-AnsibleParam -obj $params -name "domain_mode" -type "str" -choices "Win2003","Win2008","Win2008R2","Win2012","Win2012R2","WinThreshold"
```

Comment:
I'm wondering whether it is wise for us to restrict the value with these choices. There's always the possibility MS will add new modes in the future and maybe remove 2003 entirely. We can keep the docs for the values right now but after `Ensure-Prereqs` is called we can get a list of valid values by running;

```
$enum_type = (Get-Command -Name Install-ADDSForest).Parameters.DomainMode.ParameterType
$valid_modes = [Enum]::GetValues($enum_type)
```

Reply:
Yup, this is probably better. I will fix this.